onnxruntime/core/providers/cuda/tensor/split_impl.cu

```diff
@@ -8,91 +8,97 @@
-               offset;
-
-  reinterpret_cast<T*>(output_ptr[output_index])[output_pos] = input_data[id];
-}
-
-Status SplitImpl(const size_t element_size,
-                 const int block_size_including_axis_dim,
-                 const int block_size_inside_axis_dim,
-                 const int64_t* split_sizes,
-                 const int64_t* split_sizes_range,
-                 const int64_t* axis_dimension_input_output_mapping,
-                 const int num_outputs,
-                 const void* input_data,
-                 void** output_ptr,
-                 const size_t N) {
-  int blocksPerGrid = (int)(ceil(static_cast<float>(N) / GridDim::maxThreadsPerBlock));
-
-  fast_divmod block_size_including_axis_dim_div = fast_divmod(block_size_including_axis_dim);
-  fast_divmod block_size_inside_axis_dim_div = fast_divmod(block_size_inside_axis_dim);
-
-  switch (element_size) {
-    case sizeof(int8_t):
-      _SplitKernel<<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0>>>(
-          block_size_including_axis_dim_div, block_size_inside_axis_dim_div,
-          split_sizes, split_sizes_range, axis_dimension_input_output_mapping, num_outputs,
-          reinterpret_cast<const ToCudaType<int8_t>::MappedType*>(input_data),
-          output_ptr,
-          (CUDA_LONG)N);
-      break;
-    case sizeof(int16_t):
-      _SplitKernel<<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0>>>(
-          block_size_including_axis_dim_div, block_size_inside_axis_dim_div,
-          split_sizes, split_sizes_range, axis_dimension_input_output_mapping, num_outputs,
-          reinterpret_cast<const ToCudaType<int16_t>::MappedType*>(input_data),
-          output_ptr,
-          (CUDA_LONG)N);
-      break;
-    case sizeof(int32_t):
-      _SplitKernel<<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0>>>(
-          block_size_including_axis_dim_div, block_size_inside_axis_dim_div,
-          split_sizes, split_sizes_range, axis_dimension_input_output_mapping, num_outputs,
-          reinterpret_cast<const ToCudaType<int32_t>::MappedType*>(input_data),
-          output_ptr,
-          (CUDA_LONG)N);
-      break;
-    case sizeof(int64_t):
-      _SplitKernel<<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0>>>(
-          block_size_including_axis_dim_div, block_size_inside_axis_dim_div,
-          split_sizes, split_sizes_range, axis_dimension_input_output_mapping, num_outputs,
-          reinterpret_cast<const ToCudaType<int64_t>::MappedType*>(input_data),
-          output_ptr,
-          (CUDA_LONG)N);
-      break;
-    default:
-      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Type not supported for Slice operator");
+#define SPECIALIZED_IMPL(T_INT64, T_OUTPUT)                                                     \
+  template <typename T>                                                                         \
+  __global__ void _SplitKernel(const fast_divmod block_size_including_axis_dim_div,             \
```

Comment:
identifiers matching `^_[A-Z]` are reserved: https://en.cppreference.com/w/cpp/language/identifiers
consider `SplitKernel`

Reply:
i didn't see it renamed

---
In reply to: [508172886](https://github.com/microsoft/onnxruntime/pull/5541#discussion_r508172886) [](ancestors = 508172886)